### DISCUSSION TITLE
How can we run pink from its source code?

### DISCUSSION THREAD

**Discussion by dbdxnuliba** [stephane-caron/pink#99] [Q&A] [answered]
How can we install pink from its source code?

**Comment by dbdxnuliba** [comment, marked as answer]
It has been solved in ``arm_ur3.py``，
add some code like:

```py
import os
import sys
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
```

**Reply by stephane-caron** [reply]
This will work locally, assuming you have all dependencies of Pink (mainly Pinocchio) in your ``PYTHONPATH``.

It is also possible to install the library (e.g. to ``~/.local/lib``) by running the following command at the root of the repository:

```console
$ pip install --user .
```

This will take the source code in your local copy of the repository, and install it as if it were downloaded from PyPI (including any local modification you may have made).
